src/app/public/config.module.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ModuleWithProviders,
+  NgModule
+} from '@angular/core';
+
+import {
+  SkyAppConfigModuleForRootArgs
+} from './config-module-for-root-args';
+
+import {
+  SkyAppConfigHost
+} from './host';
+
+import {
+  SkyAppParamsConfig
+} from './params-config';
+
+@NgModule({})
+export class SkyAppConfigModule {
+
+  public static forRoot(
+    config: SkyAppConfigModuleForRootArgs = {}
+  ): ModuleWithProviders<SkyAppConfigModule> {
+
+    return {
+      ngModule: SkyAppConfigModule,
+      providers: [
+        {
+          provide: SkyAppParamsConfig,
+          useFactory() {
+            return new SkyAppParamsConfig({
+              params: config.params
+            });
+          }
+        },
+        {
+          provide: SkyAppConfigHost,
```

Comment:
Are we happy with the name of this provider (note the name of `SkyAppParamsConfig` above)?

Reply:
I feel like we may have discussed this before, but what's the reason we went with `SkyAppParamsConfig` and not `SkyAppConfigParams`?

Reply:
I don't think we caught the name discrepancy in the last PR, unfortunately.